mypy/checkmember.py

```diff
@@ -245,6 +245,15 @@ def analyze_member_var_access(name: str, itype: Instance, info: TypeInfo,
                     getattr_type = expand_type_by_instance(bound_method, typ)
                     if isinstance(getattr_type, CallableType):
                         return getattr_type.ret_type
+        else:
+            setattr_func = info.get_method('__setattr__')
+            if setattr_func and setattr_func.info.fullname() != 'builtins.object':
+                setattr_f = function_type(setattr_func, builtin_type('builtins.function'))
+                bound_type = bind_self(setattr_f, original_type)
+                typ = map_instance_to_supertype(itype, setattr_func.info)
+                setattr_type = expand_type_by_instance(bound_type, typ)
+                if isinstance(setattr_type, CallableType):
+                    return setattr_type.arg_types[-1]
```

Comment:
Ah, yes. Then it might be a good idea to check the signature as is done with `__getattr__` anyway. because any non-valid signature should be rejected.